### ISSUE TITLE
[Question] Widgets

### ISSUE THREAD

**Issue by danimesq** [capyloon/nutria#27]
Are there a documentation about developing widgets? Are 3rd-party widgets supported?

**Comment by danimesq** [comment]
> Currently widgets are similar to apps, only loaded in a smaller container on the homescreen. If you check the ones in [apps/widgets/](https://github.com/capyloon/nutria/tree/main/apps/widgets) you can see how to implement them.

I love the idea that apps and widgets could share the same codebase.
So, there could be presets (such as sizes) and apps could adapt into their widget modes/forms.

**Comment by fabricedesre** [comment]
Yes, absolutely. And an app could include a widget view in its own URL, calling the yet to be implemented activity to add it to the homescreen.

Maybe we could also support either manifest declaration of widgets, and a <link rel="widget" > to discover them in web pages.

**Comment by danimesq** [comment]
> Yes, absolutely. And an app could include a widget view in its own URL, calling the yet to be implemented activity to add it to the homescreen.

The app itself calling to add its widget is quite interesting; personally haven't seen such behavior in apps before yet.

Also, Nutria could read the manifest.webapp of all apps, then form a list of widgets; so, when "right-clicking" in the Launcher, user will get a list of widgets available to add.
